src/Training.AirBnb.Clone.Backend/AirBnB.Api/Configurations/HostConfiguration.Extensions.cs

```diff
@@ -1,10 +1,85 @@
+using AirBnB.Persistence.DataContexts;
+using AirBnB.Persistence.Repositories;
+using AirBnB.Persistence.Repositories.Interfaces;
+using FluentValidation;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Migrations;
 
 namespace AirBnB.Api.Configurations;
 
 public static partial class HostConfiguration
 {
+    private static readonly ICollection<Assembly> Assemblies;
+
+    static HostConfiguration()
+    {
+        Assemblies = Assembly.GetExecutingAssembly().GetReferencedAssemblies().Select(Assembly.Load).ToList();
+        Assemblies.Add(Assembly.GetExecutingAssembly());
+    }
+
+    /// <summary>
+    /// Registers NotificationDbContext in DI 
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <returns></returns>
+    private static WebApplicationBuilder AddNotificationInfrastructure(this WebApplicationBuilder builder)
+    {
+        builder.Services.AddDbContext<NotificationDbContext>(options =>
+            options.UseNpgsql(builder.Configuration.GetConnectionString("DefaultConnection"),
+                o => o.MigrationsHistoryTable(
+                    tableName: HistoryRepository.DefaultTableName,
+                    schema: "notification")));
+        return builder;
+    }
+
+    /// <summary>
+    /// Configures IdentityInfrastructure including controllers
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <returns></returns>
+    private static WebApplicationBuilder AddIdentityInfrastructure(this WebApplicationBuilder builder)
+    {
+        builder.Services.AddDbContext<IdentityDbContext>(options =>
+            options.UseNpgsql(builder.Configuration.GetConnectionString("DefaultConnection"),
+                o => o.MigrationsHistoryTable(
+                    tableName: HistoryRepository.DefaultTableName,
+                    schema: "identity")));
+
+        builder.Services
+            .AddScoped<IUserRepository, UserRepository>()
+            .AddScoped<IUserService, UserService>();
+
+        builder.Services.Configure<ValidationSettings>(builder.Configuration.GetSection(nameof(ValidationSettings)));
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Configures the Dependency Injection container to include validators from referenced assemblies.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <returns></returns>
+    private static WebApplicationBuilder AddValidators(this WebApplicationBuilder builder)
+    {
+        builder.Services.AddValidatorsFromAssemblies(Assemblies);
+        return builder;
+    }
+
+    /// <summary>
+    /// Configures AutoMapper for object-to-object mapping using the specified profile.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <returns></returns>
+    private static WebApplicationBuilder AddMappers(this WebApplicationBuilder builder)
+    {
+        builder.Services.AddAutoMapper(Assemblies);
+        return builder;
+    }
```

Comment:
Move these methods before `AddNotificationInfrastructure` method and after `HostConfiguration` property